app/views/security_group/_fw_rules.haml

```diff
@@ -0,0 +1,80 @@
+%h3
+  = _('Firewall Rules')
+  %table.table.table-bordered
+    %thead
+      %tr
+        %th
+          = _('Network Protocol')
+        %th
+          = _('Host Protocol')
+        %th
+          = _('Direction')
+        %th
+          = _('Remote Security Group (name - ref)')
+        %th
+          = _('IP Range')
+        %th
+          = _('Port')
+        %th
+          = _('End Port')
+        %th
+          = _('Action')
+    %tbody
+      %tr{"ng-repeat" => "fw_rule in vm.securityGroupModel.firewall_rules",
+          "ng-form"   => "rowForm",
+          "ng-if"     => "!fw_rule.deleted"}
+        %td.table-view-pf-select
+          %select{"name"                        => "network_protocol",
+            "data-width"                  => "auto",
+            "ng-model"                    => "vm.securityGroupModel.firewall_rules[$index].network_protocol",
+            "ng-options"                  => "network_protocol for network_protocol in vm.networkProtocols",
+            "ng-selected"                 => "fw_rule.network_protocol",
+            "selectpicker-for-select-tag" => ""}
+        %td.table-view-pf-select
+          %select{"name"                        => "host_protocol",
+            "data-width"                  => "auto",
+            "ng-model"                    => "vm.securityGroupModel.firewall_rules[$index].host_protocol",
+            "ng-options"                  => "host_protocol for host_protocol in vm.hostProtocols",
+            "ng-selected"                 => "fw_rule.host_protocol",
+            "selectpicker-for-select-tag" => ""}
+        %td.table-view-pf-select
+          %select{"name"                        => "direction",
+            "data-width"                  => "auto",
+            "ng-model"                    => "vm.securityGroupModel.firewall_rules[$index].direction",
+            "ng-options"                  => "direction for direction in vm.directions",
+            "ng-selected"                 => "fw_rule.direction",
+            "selectpicker-for-select-tag" => ""}
+        %td.table-view-pf-select
+          %select{"name"                        => "source_security_group_id",
+            "data-width"                  => "auto",
+            "ng-model"                    => "vm.securityGroupModel.firewall_rules[$index].source_security_group_id",
```

Comment:
I noticed that `vm.securityGroupModel.firewall_rules[$index].source_security_group_id` is `undefined`... the API does not seem to populate it. Can you verify?